BitmarkRegistry/Services/ChibitronicsService.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  ChibitronicsService.swift
+//  BitmarkRegistry
+//
+//  Created by Macintosh on 8/1/19.
+//  Copyright © 2019 thuyentruong. All rights reserved.
+//
+
+import Foundation
+import BitmarkSDK
+
+enum VerificationLinkSource {
+  case qrCode, deepLink
+}
+
+struct ChibitronicsService {
+
+  let verificationLink: String
+  let source: VerificationLinkSource
+  lazy var separator: Character = { source == .qrCode ? "|" : "/" }()
+  lazy var regex: String = { "(\\w+ ){5}\\w+\\\(separator)https?://.+" }()
+
+  var code: String!
+  var url: URL!
+
+  init(verificationLink: String, source: VerificationLinkSource) {
+    self.verificationLink = verificationLink
+    self.source = source
+  }
+
+  mutating func isValid() -> Bool {
+    return verificationLink.range(of: regex, options: .regularExpression, range: nil, locale: nil) != nil
+  }
+
+  mutating func extractData() -> (code: String, url: URL)? {
+    let codeParts = verificationLink.split(separator: separator, maxSplits: 1)
+
+    if let url = URL(string: String(codeParts[1])) {
+      self.code = String(codeParts[0])
+      self.url = url
+      return (code: code, url: url)
+    } else {
+      return nil
+    }
+  }
+
+  func requestAuthorization(for account: Account, completion: @escaping (Error?) -> Void) throws {
+    guard let url = url, let code = code else { return }
+    let message = "Verify|\(code)"
+    let signature = try account.sign(message: message.data(using: .utf8)!)
+
+    let infoLog = "requestAuthorization for account: \(account.getAccountNumber()); code: \(code)"
+    ErrorReporting.breadcrumbs(info: infoLog, category: "Chibitronics")
+    Global.log.info(infoLog)
+
+    let params: [String : String] = [
+      "bitmark_account": account.getAccountNumber(),
+      "code": code,
+      "signature": signature.hexEncodedString
+    ]
+    let jsonData = try JSONSerialization.data(withJSONObject: params)
```

Comment:
IMHO, we should get rid of `JSONSerialization` as much as possible. `[String: String]` can be handled by `JSONEncoder` instead, so the code looks to be more "swifty".